src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1715,6 +1724,81 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
             return true;
         }
 
+        // Reload's the dummy nodes and serializes the new node information once the package resolves it. 
+        private void ReloadDummyNodes()
+        {
+            JObject dummyNodeJSON = null;
+            WorkspaceModel currentWorkspace = this.CurrentWorkspace;
+
+            if (currentWorkspace == null || string.IsNullOrEmpty(currentWorkspace.FileName))
+            {
+                return;
+            }
+
+            String filePath = Path.GetFullPath(currentWorkspace.FileName);
+
+            CustomNodeManager.AddUninitializedCustomNodesInPath(
+                        Path.GetDirectoryName(filePath),
+                        IsTestMode);
+
+            // Get the dummy node's in the current workspace. 
+            var dummyNodes = currentWorkspace.Nodes.OfType<DummyNode>();
+
+            foreach (DummyNode dn in dummyNodes)
+            {
+                dummyNodeJSON = dn.OriginalNodeContent as JObject;
+
+                if (dummyNodeJSON != null)
+                {
+                    // Deserializing the dummy node and verifying if it is resolved by the downloaded package
+                    NodeModel resolvedNode = WorkspaceModel.GetNodeModelForDummyNode(
+                                                               dummyNodeJSON.ToString(),
+                                                               LibraryServices,
+                                                               EngineController,
+                                                               Scheduler,
+                                                               NodeFactory,
+                                                               IsTestMode,
+                                                               false,
+                                                               CustomNodeManager);
+
+                    if (!(resolvedNode is DummyNode))
+                    {
+                        currentWorkspace.SetNodeViewDataOnResolvedNode(dn, resolvedNode);
+
+                        // Disable Run's temporarlity while the dummy node is removed and the resolved node is added. 
+                        EngineController.DisableRun = true;
+                        currentWorkspace.RemoveAndDisposeNode(dn);
+                        currentWorkspace.AddAndRegisterNode(resolvedNode, false);
+
+                        // Adding back the connectors for the resolved node. 
+                        List<ConnectorModel> connectors = dn.AllConnectors.ToList();
+                        foreach (var connectorModel in connectors)
+                        {
+                            var startNode = connectorModel.Start.Owner;
+                            var endNode = connectorModel.End.Owner;
+
+                            if (startNode is DummyNode && startNode.Name == resolvedNode.Name)
```

Comment:
what is this bit about?

Reply:
When I am adding back the connectors, if the startNode or endNode was a dummy node(and it is being replaced with the resolved node), then the new connector should be added on the resolved node. Since the graph could have many dummy nodes, I need to check the GUID before setting the startNode and endNode for the new connector. I have changed this check the GUID property.